10up/distributor
# Reduce duplication pushing posts.

### Description of the Change


Replaces duplicate code in the network and external connection classes by using the post abstraction.


Closes #997 

### How to test the Change


This requires testing on both **network (internal)** connections and **external** connections. The steps for each are the same:

* Create a post on the source site, test with
  * terms
  * content
  * media
  * post meta (this mini [plugin updates an item on each save](https://gist.github.com/peterwilsoncc/61eeb5f2337cdf4976608a5123acbb1d))
* ensure that push as draft/push as published feature works as expected
* ensure the linking, unlinking of posts works as expected (including updates while unlinked)
* ensure time stamp updates are unchanged from the develop branch (including modified times, this may require looking in the database)
* using xdebug to compare the payloads from develop and this branch would be lovely too.


### Changelog Entry

> Changed - Reduced duplicate code pushing posts to connections.


### Credits

Props @peterwilsoncc, @cadic.


### Checklist:
